hls4ml/model/types.py

```diff
@@ -237,20 +81,36 @@ class IntegerPrecisionType(PrecisionType):
 
     def __init__(self, width=16, signed=True):
         super().__init__(width=width, signed=signed)
-        self.integer = width
-        self.fractional = 0
 
     def __str__(self):
         typestring = '{signed}int<{width}>'.format(signed='u' if not self.signed else '', width=self.width)
         return typestring
 
     def __eq__(self, other):
-        eq = self.width == other.width
-        eq = eq and self.signed == other.signed
-        # These are probably unnecessary
-        eq = eq and self.integer == other.integer
-        eq = eq and self.fractional == other.fractional
-        return eq
+        if isinstance(other, IntegerPrecisionType):
+            return super().__eq__(other)
+
+        return False
+
+    @property
+    def integer(self):
+        return self.width
+
+    @property
+    def fractional(self):
+        return 0
+
+    @property
+    def rounding_mode(self):
+        return RoundingMode.TRN
+
+    @property
+    def saturation_mode(self):
+        return SaturationMode.WRAP
+
+    @property
+    def saturation_bits(self):
+        return None
```

Comment:
Should this be 0 instead of None?

Reply:
yes. good catch! i left it while debugging :man_facepalming: